src/arch/src/aarch64/fdt.rs

```diff
@@ -338,17 +338,27 @@ fn create_cpu_nodes(fdt: &mut Vec<u8>, vcpu_mpidr: &[u64]) -> Result<()> {
             // Please check out
             // https://github.com/devicetree-org/devicetree-specification/releases/download/v0.3/devicetree-specification-v0.3.pdf,
             // section 3.8.
-            append_property_u32(fdt, cache.type_.of_cache_size(), cache.size_ as u32)?;
-            append_property_u32(
-                fdt,
-                cache.type_.of_cache_line_size(),
-                cache.line_size as u32,
-            )?;
-            append_property_u32(
-                fdt,
-                cache.type_.of_cache_sets(),
-                cache.number_of_sets as u32,
-            )?;
+            if cache.size_.is_some() {
+                append_property_u32(
+                    fdt,
+                    cache.type_.of_cache_size(),
+                    cache.size_.unwrap() as u32,
+                )?;
+            }
```

Comment:
nit: As this is used multiple times below (and in similar fashion), maybe this can be a function or macro for deduplication and to avoid the use of `unwrap` with an `if let`.